### PR TITLE
infra: update to a newer xcode version to get out ahead of deprecations

### PR DESCRIPTION
**Pull request by abernix**
This updates to Xcode 13.4 (currently, 13.4.1, according to the [matrix] of supported versions).  This should get us out ahead of the deprecation policy that is defined in the [CircleCI Xcode Policy].

[CircleCI Xcode Policy]: https://circleci.com/docs/xcode-policy#xcode-patches
[matrix]: https://circleci.com/docs/using-macos#supported-xcode-versions

### CODE REVIEW

**Review by garypen — APPROVED** [review]

**Comment by abernix** [comment]
> @abernix your pull request is missing a changelog!

I will add the CHANGELOG after this merges to `main` and then into the release PR because I don't want to wait for CI to run another time with the CHANGELOG change.

**Comment by abernix** [comment]
> > @abernix your pull request is missing a changelog!
> 
> I will add the CHANGELOG after this merges to `main` and then into the release PR because I don't want to wait for CI to run another time with the CHANGELOG change.

Done on #1707

**Pull request merged**
